src/main/drivers/pwm_output_hal.c

```diff
@@ -0,0 +1,245 @@
+/*
```

Comment:
Can't this file be unified with current implementation of `pwm_output.c`? Most of the code stays the same anyway and it's a pain to maintain two almost identical files.